core/src/indexer/native_transfer.rs

```diff
@@ -0,0 +1,130 @@
+use std::{str::FromStr, sync::Arc};
+
+use colored::Colorize;
+use ethers::{
+    abi::Hash,
+    types::{Action, Address, Bytes, U256, U64},
+};
+use futures::future::try_join_all;
+use serde::Serialize;
+use serde_json::json;
+use tracing::{error, info};
+
+use super::start::StartIndexingError;
+use crate::{
+    event::{callback_registry::TxInformation, EventMessage},
+    provider::JsonRpcCachedProvider,
+    streams::StreamsClients,
+};
+
+#[derive(Serialize)]
+pub struct NativeTransfer {
+    pub from: Address,
+    pub to: Address,
+    pub value: U256,
+    pub transaction_information: TxInformation,
+}
+
+pub async fn native_transfer_block_consumer(
+    provider: Arc<JsonRpcCachedProvider>,
+    block_numbers: &[U64],
+    network_name: &str,
+    streams_client: Option<&StreamsClients>,
+) -> Result<(), StartIndexingError> {
+    let trace_futures: Vec<_> = block_numbers.iter().map(|n| provider.trace_block(*n)).collect();
+
+    let trace_calls = try_join_all(trace_futures).await?;
+
+    let native_transfers = trace_calls
+        .into_iter()
+        .flatten()
+        .filter_map(|trace| {
+            let action = match trace.action {
+                Action::Call(call) => Some(call),
+                _ => None,
+            }?;
+
+            let has_value = !action.value.is_zero();
+            let no_input = action.input == Bytes::from_str("0x").unwrap();
+            let is_native_transfer = has_value && no_input;
+
+            if is_native_transfer {
+                if trace.transaction_hash.is_none() {
+                    error!("Transaction hash should exist for block trace {}", trace.block_number);
+                    return None;
+                }
+
+                Some(NativeTransfer {
+                    from: action.from,
+                    to: action.to,
+                    value: action.value,
+                    transaction_information: TxInformation {
+                        network: network_name.to_owned(),
+                        address: Address::zero(),
```

Comment:
you dont know the contract address this did the action from if present? could be useful say "get me all the eth transfers which were sent from x contract" or something

Reply:
Hmm, so would you suggest making this `address` equal to the `transfer.from` address in this case?